### PR TITLE
NW6 | Zeliha Pala | React-Module-Project | Week-3 | Feature/add booking form

### PR DESCRIPTION
**Pull request by zelihapala**
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

**Review by fikretellek — APPROVED** [review]
good job!

**Review by fikretellek — APPROVED** [review]
well done!

File: src/components/AddBookingForm/AddBookingForm.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    setFormData({
+      id: "",
+      firstName: "",
+      surname: "",
+      email: "",
+      title: "",
+      roomId: "",
+      checkInDate: "",
+      checkOutDate: "",
+    });
```
Comment (Ara225) [inline src/components/AddBookingForm/AddBookingForm.jsx:39]: This object initalization is done twice with the same data. It works, but when you see code repeated like this it's worth thinking about whether you could put it inside a function - makes for tider code :)

File: src/components/AddBookingForm/AddBookingForm.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        value={formData.checkOutDate}
+        onChange={handleChange}
+      />
+      <input
+        type="text"
+        name="roomId"
+        placeholder="Room ID"
+        value={formData.roomId}
+        onChange={handleChange}
+      />
```
Comment (Ara225) [inline src/components/AddBookingForm/AddBookingForm.jsx:100]: This works but there's a tider way to do this for next time, have a look at https://legacy.reactjs.org/docs/lists-and-keys.html

File: src/components/AddBookingForm/AddBookingForm.test.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    fireEvent.change(getByPlaceholderText('First Name'), { target: { value: 'John' } });
+    fireEvent.change(getByPlaceholderText('Surname'), { target: { value: 'Doe' } });
+    fireEvent.change(getByPlaceholderText('Title'), { target: { value: 'Mr.' } });
+    fireEvent.change(getByPlaceholderText('Email'), { target: { value: 'john.doe@example.com' } });
+    fireEvent.change(getByPlaceholderText('Check In Date'), { target: { value: '2024-03-15' } });
+    fireEvent.change(getByPlaceholderText('Check Out Date'), { target: { value: '2024-03-17' } });
+    fireEvent.change(getByPlaceholderText('Room ID'), { target: { value: '123' } });
+
+    fireEvent.click(getByText('Book now'));
+  });
```
Comment (Ara225) [inline src/components/AddBookingForm/AddBookingForm.test.jsx:46]: Love your tests :)

**Review by Ara225 — COMMENTED** [review]
I've included some ideas for refactoring that are worth being aware of but it looks good otherwise

**Pull request merged**